### ISSUE TITLE
Monoisome

### ISSUE THREAD

**Issue by larsenwork** [larsenwork/monoid#98]
I've just realised http://fontawesome.io uses the same grid as Monoid. This would make it fairly easy to create a version of Monoid that includes all the Font-Awesome icons as ligatures. 

This would obviously increase the file size too much for it to be part of the default Monoid font. It would look something like this: 

![screen shot 2015-07-30 at 13 07 51](https://cloud.githubusercontent.com/assets/3859425/8982152/ad6396fe-36be-11e5-8681-e46c3a1445d6.png)

I've asked @davegandy on twitter so awaiting his reply. https://twitter.com/larsenwork/status/626711927989895168

`¯\_(ツ)_/¯`

**Comment by larsenwork** [comment]
Brackets and Atom both currently have problems with calculating line length but this will hopefully be fixed soon: 

![screen shot 2015-07-30 at 13 59 29](https://cloud.githubusercontent.com/assets/3859425/8982692/fd37e8b0-36c3-11e5-8f62-f3e01101e65d.png)
![screen shot 2015-07-30 at 14 02 24](https://cloud.githubusercontent.com/assets/3859425/8982693/fd38d630-36c3-11e5-90aa-0c2471a117c0.png)

(I'm disabling ligatures for cursor line in Atom which is why there's no 👍 there

**Comment by mikebronner** [comment]
I LOVE this!!!!! (haven't installed it yet, but the idea is genius) Makes the code so much more expressive -- you see right away which icon you get.

**Comment by larsenwork** [comment]
@mikebronner yep, I need some help to make this happen though as I don't have the time right now. Wanna help? It's very easy and should only take an hour or two.

**Comment by larsenwork** [comment]
You just need to download Fontforge to see the glyph names 

![screen shot 2015-07-30 at 18 14 05](https://cloud.githubusercontent.com/assets/3859425/8988606/7b3247ce-36e7-11e5-84a0-b73d51be4d60.png)

and then write them in the text file

![screen shot 2015-07-30 at 18 18 06](https://cloud.githubusercontent.com/assets/3859425/8988605/7b0f9940-36e7-11e5-92c7-5911219b3f9c.png)

**Comment by mikebronner** [comment]
Absolutely would love to help will start when I get back to the office tonight

**Comment by larsenwork** [comment]
Perfect, I'll upload the files in a minute.

**Comment by larsenwork** [comment]
Some notes:

I'll created a branch that we are working in and then we can merge to master once it's done. 
The text file is called a feature file and can be edited in any code/text editor.

Fontforge: http://fontforge.github.io

The format (sfdir) is a folder so you open it in FontForge not by double clicking but by selecting and clicking "OK".

Sometimes an icon has two aliases e.g. indent_left which is both fa-dedent and fa-outdent
```
  sub f a hyphen d e d e n t by indent_left;
  sub f a hyphen o u t d e n t by indent_left;
```

I've removed all Monoid ligatures - I'll re-add them once the Awesome part is working.

I've switched to "compact" encoding - I'll switch back to UTF-8 once the Awesome part is working.

I've added and adjusted all the icons, written the first 100+ ligatures so what is left is to write the rest of the glyph names in the .fea file.

Don't be too thorough - I can quickly see if there are any errors when importing the feature file.

And please don't hesitate to ask if you have any questions!

**Comment by larsenwork** [comment]
I've also had to rename e.g. plus, minus and asterisk as those names are already used in a normal font.

**Comment by larsenwork** [comment]
All yours: https://github.com/larsenwork/monoid/blob/Monoisome/Monoisome/Monoisome.fea#L117

Just generated the font to test the size: 230kb so definitely not going in the build scripts.

**Comment by mikebronner** [comment]
Regarding the naming: I would recommend including the fa- in the name, so as not to conflict. Especially if we add Glyphicons, or other icon sets that are commonly used.

**Comment by larsenwork** [comment]
Good idea - let's do that if/when we add other icon fonts :+1: 

![screen shot 2015-07-31 at 11 04 30](https://cloud.githubusercontent.com/assets/3859425/9005025/088fc2fc-377b-11e5-8aa2-ee07d1ab1a62.png)

**Comment by larsenwork** [comment]
Closing this for now, as it seems to work well

**Comment by mikebronner** [comment]
Awesome! Is it available in the latest version already?

**Comment by larsenwork** [comment]
website

**Comment by mikebronner** [comment]
Got it ... so to use it effectively, this replaces the Monoid-Regular.ttf, right? (Otherwise I don't have bold or italics if I leave it named Monoisome).

**Comment by larsenwork** [comment]
I'd image most editors that support ligatures also let you pick different font families for bold/italic?

**Comment by mikebronner** [comment]
Unfortunately that doesn't seem to be the case with any of the editors I tried.

**Comment by mamartins** [comment]
I can confirm that on VScode and Atom the problem still occurs :(

**Comment by boazblake** [comment]
Have only tried in atom - can not get monoisome font or the equality lit's to work either

**Comment by larsenwork** [comment]
Ok, not sure how much time I'll spend on fixing it though since I haven't used font-icons in over a year and maintaining something like font awesome that luckily very few still use... I'd rather focus on creating new and better stuff :)

**Comment by mikebronner** [comment]
I think font-awesome has one of the best adoptions, seeing how well their FontAwesome 5 Kickstarter campaign went. But yes, there are more and more alternatives popping up all the time. (I'm just chiming in here because I was helping with this. I have to admit I have not been using Monoisome either -- I do use FontAwesome on a daily basis, but regrettably I have been using a different font for development use.)

**Comment by boazblake** [comment]
I am using fira-code for now - but I am very interested in checking out
Monoid and Monoisome

**Comment by mamartins** [comment]
I'm still using this font :)

**Comment by Samoht1** [comment]
I'd love a fix to use bold with monoisome :)
